agent/command/git_merge_pr.go

```diff
@@ -0,0 +1,168 @@
+package command
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/agent/internal"
+	"github.com/evergreen-ci/evergreen/agent/internal/client"
+	"github.com/evergreen-ci/evergreen/thirdparty"
+	"github.com/evergreen-ci/evergreen/util"
+	"github.com/evergreen-ci/utility"
+	"github.com/google/go-github/github"
+	"github.com/mitchellh/mapstructure"
+	"github.com/mongodb/grip/level"
+	"github.com/mongodb/grip/message"
+	"github.com/mongodb/grip/recovery"
+	"github.com/mongodb/grip/send"
+	"github.com/pkg/errors"
+)
+
+const (
+	// valid Github merge methods
+	githubMergeMethodMerge  = "merge"
+	githubMergeMethodSquash = "squash"
+	githubMergeMethodRebase = "rebase"
+
+	GithubContext = "evergreen/commitqueue"
+)
+
+type gitMergePr struct {
+	Status string `mapstructure:"status"`
+	URL    string `mapstructure:"url"`
+	Token  string `mapstructure:"token"`
+
+	statusSender send.Sender
+	base
+}
+
+func gitMergePrFactory() Command   { return &gitMergePr{} }
+func (c *gitMergePr) Name() string { return "git.merge_pr" }
+
+func (c *gitMergePr) ParseParams(params map[string]interface{}) error {
+	err := mapstructure.Decode(params, c)
+	if err != nil {
+		return err
+	}
+
+	if len(c.Status) == 0 {
+		return errors.New("status must be specified")
+	}
+
+	return nil
+}
+
+func (c *gitMergePr) Execute(ctx context.Context, comm client.Communicator, logger client.LoggerProducer, conf *internal.TaskConfig) error {
+	var err error
+	defer func() {
+		pErr := recovery.HandlePanicWithError(recover(), nil, "unexpected error in git.merge_pr")
+		status := evergreen.MergeTestSucceeded
+		if err != nil || pErr != nil {
+			status = evergreen.MergeTestFailed
+		}
+		logger.Task().Error(err)
+		logger.Task().Critical(pErr)
+		logger.Task().Error(comm.ConcludeMerge(ctx, conf.Task.Version, conf.ProjectRef.Identifier, status))
+	}()
+	if err = util.ExpandValues(c, conf.Expansions); err != nil {
+		return errors.Wrap(err, "can't apply expansions")
+	}
+
+	patchDoc, err := comm.GetTaskPatch(ctx, client.TaskData{ID: conf.Task.Id, Secret: conf.Task.Secret})
+	if err != nil {
+		return errors.Wrap(err, "unable to get patch")
+	}
+
+	_, projectToken, err := getProjectMethodAndToken(c.Token, conf.Expansions.Get(evergreen.GlobalGitHubTokenExpansion), conf.Distro.CloneMethod)
+	if err != nil {
+		return errors.Wrap(err, "failed to get token")
+	}
+	httpClient := utility.GetOAuth2HTTPClient(projectToken)
+	defer utility.PutHTTPClient(httpClient)
+	githubClient := github.NewClient(httpClient)
+
+	c.statusSender, err = send.NewGithubStatusLogger("evergreen", &send.GithubOptions{
+		Token: projectToken,
+	}, "")
+	if err != nil {
+		return errors.Wrap(err, "failed to setup github status logger")
+	}
+
+	// TODO: should not get here for failed patches
```

Comment:
Is this a todo for a future part or something that should be added for this? It seems a patch status check could happen before 78

Reply:
it's for a future ticket, but I went ahead and did part of it now because the part I'm working on next is easier because of it. The comment here was actually wrong - this command should run regardless of if the commit queue entry fails. We need to report a ❌  or ✅  to github, but only merge if it passed. I added a different todo to create a new field on the patch data model that says if the rest of the patch passed